bambi/backend/pymc.py

```diff
@@ -203,43 +221,27 @@ def _run_mcmc(
                             random_seed=random_seed,
                             **kwargs,
                         )
+                        idata_from = "pymc"
                     else:
                         raise
-            elif sampler_backend == "nuts_numpyro":
-                import pymc.sampling_jax  # pylint: disable=import-outside-toplevel
-
-                if not chains:
-                    # sample_numpyro_nuts does not handle chains = None like pm.sample does
-                    chains = 4
-                idata = pymc.sampling_jax.sample_numpyro_nuts(
-                    draws=draws,
-                    tune=tune,
-                    chains=chains,
-                    random_seed=random_seed,
-                    **kwargs,
-                )
-            elif sampler_backend == "nuts_blackjax":
-                import pymc.sampling_jax  # pylint: disable=import-outside-toplevel
-
-                # sample_blackjax_nuts does not handle chains = None like pm.sample does
-                if not chains:
-                    chains = 4
-                idata = pymc.sampling_jax.sample_blackjax_nuts(
-                    draws=draws,
-                    tune=tune,
-                    chains=chains,
-                    random_seed=random_seed,
-                    **kwargs,
-                )
-            else:
-                raise ValueError(
-                    f"sampler_backend value {sampler_backend} is not valid. Please choose one of"
-                    f"'mcmc', 'nuts_numpyro' or 'nuts_blackjax'"
-                )
-        idata = self._clean_results(idata, omit_offsets, include_mean)
+        elif sampler_backend in BAYEUX_SAMPLERS:
+            import bayeux as bx
+            import jax
+
+            bx_model = bx.Model.from_pymc(self.model)
+            bx_sampler = getattr(bx_model.mcmc, sampler_backend)
+            idata = bx_sampler(seed=jax.random.key(0), **kwargs)
```

Comment:
there's a `random_seed` that gets passed in -- I presume from `numpy`? We could use `jax.random.key(np.random.randint(2**32 - 1, seed=random_key))`, which I think maintains the expected entropy?

Reply:
The `random_seed` parameter of `model.fit` is an int or list of ints (if cores > 1). `np.random.randint` doesn't have a `seed` param so it will generate a random in each time. Did you perhaps mean to use `jax.random.randint`?

In the case of JAX based methods, we could do something like

```python3
            import bayeux as bx
            import jax
            
            # Seed is required for bayeux
            if random_seed is None:
                random_seed = 0

            bx_model = bx.Model.from_pymc(self.model)
            bx_sampler = operator.attrgetter(sampler_backend)(bx_model.mcmc)
            idata = bx_sampler(seed=jax.random.key(random_seed), **kwargs)
```

although the name `random_seed` is ambiguous because it is not random. 

Since JAX random functions explicitly consume the state (the key), feeding the same key to a random function will always result in the same sample being generated. If the user does not pass a key, we explicitly set it to `0`, else the user passed key is passed to `jax.random_key`. This way, we get reproducible results. 

Apologies for the newb question, I am relatively new to JAX PRNG.

Reply:
Hah. Thanks! I got it mixed up.